src/pointer-analysis/value_set_dereference.cpp

```diff
@@ -30,6 +31,30 @@ Author: Daniel Kroening, kroening@kroening.com
 #include <util/simplify_expr.h>
 #include <util/ssa_expr.h>
 
+/// Returns true if \p expr is complicated enough that a local definition (using
+/// a let expression) is preferable to repeating it, potentially many times.
+/// Of course this is just a heuristic -- currently we allow any expression that
+/// only involves one symbol, such as "x", "(type*)x", "x[0]" (but not "x[y]").
+/// Particularly we want to make sure to insist on a local definition of \p expr
+/// is a large if-expression, such as `p == &o1 ? o1 : p == &o2 ? o2 : ...`, as
+/// can result from dereferencing a subexpression.
+static bool should_use_local_definition_for(const exprt &expr)
```

Comment:
At least the implementation could just be replaced by using `has_subexpr(ID_symbol)`, but maybe you want to remove the function altogether.

Reply:
No, this looks for *multiple* symbol expressions, looking to permit `x`, `(int*)x` etc but not `x[y]` or `*(x + (y ? z : w))` or similar